### PR TITLE
MS-52: Clients-API-REST

### PR DESCRIPTION
**Pull request by NicoMunoz909**
- Add new model
- Add endpoints + controllers using GET, POST, PUT & DELETE
- Import Postman collection
- Validate data using mongoose schemas validation

### CODE REVIEW

File: src/controllers/clients.js [hunk 1]
```diff
@@ -1,117 +1,115 @@
...
+    email: req.body.email,
+    phoneNumber: req.body.phoneNumber,
+    cuit: req.body.cuit,
+    address: req.body.address,
+    activity: req.body.activity,
   };
-  clients.push(newClient);
-  res.status(201).json(clients);
+  Clients.create(newClient);
+  res.status(201).json(newClient);
```
Comment (nicolasblasco) [inline src/controllers/clients.js:69]: @NicoMunoz909 we should use a callback or a promise to return an error response. We could use the same promise used in getAll.

File: src/models/clients.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  address: {
+    type: String,
+    required: true,
+    maxlength: 50,
+  },
+  activity: {
+    type: String,
+    required: true,
+    maxlength: 20,
```
Comment (nicolasblasco) [inline src/models/clients.js:32]: @NicoMunoz909 we should write this property in camelCase --> maxLength

File: src/controllers/clients.js [hunk 3]
```diff
@@ -1,117 +1,115 @@
...
 
 //CLIENT LIST
 const getAll = (req, res) => {
-  return res.status(200).json(clients);
+  Clients.find()
+    .then((clients) => {
+      return res.status(200).json(clients);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/clients.js:10]: @NicoMunoz909  we should change this HTTP response status code to 404: Not Found.

File: src/controllers/clients.js [hunk 4]
```diff
@@ -1,117 +1,115 @@
...
-    (client) => client.activity === req.params.activity
-  );
-  if (client) return res.status(200).json(client);
-  return res.status(404).send("Client not found");
+  Clients.findById(req.params.id)
+    .then((client) => {
+      return res.status(200).json(client);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/clients.js:20]: @luisnannini we should change this HTTP response status code to 404: Not Found.

File: src/models/clients.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-useless-escape */
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const ClientsSchema = new Schema({
+  name: {
+    type: String,
+    required: true,
+    minLength: 6,
```
Comment (nicolasblasco) [inline src/models/clients.js:9]: @NicoMunoz909 we should remove minLength or lower the value. Example: Tesla wouldn't be valid.

File: src/models/clients.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    maxLength: 11,
+    match: /^\d+$/,
+  },
+  address: {
+    type: String,
+    required: true,
+    minLength: 6,
+    maxLength: 50,
+    match: /^[a-zA-Z0-9\s,'-]*$/,
```
Comment (nicolasblasco) [inline src/models/clients.js:41]: @NicoMunoz909 we shouldn't validate it an address has a .
For example,  Av. Pellegrini is invalid. We should correct that.

**Review by JamirMansilla — APPROVED** [review]
👍

**Pull request merged**